tests/test-cases/Group6-VIC-Machine/6-07-Create-Network.robot

```diff
@@ -576,8 +576,49 @@ Container network invalid 1
 Container network invalid 2
     Pass execution  Test not implemented
 
-Container network 1
-    Pass execution  Test not implemented
+Reset VCH doesn't cause unintentionally exposed ports from container network
+    Set Test Environment Variables
+    # Attempt to cleanup old/canceled tests
+    Run Keyword And Ignore Error  Cleanup Dangling VMs On Test Server
+    Run Keyword And Ignore Error  Cleanup Datastore On Test Server
+
+    ${output}=  Run  bin/vic-machine-linux create --name=%{VCH-NAME} --target="%{TEST_USERNAME}:%{TEST_PASSWORD}@%{TEST_URL}" --thumbprint=%{TEST_THUMBPRINT} --image-store=%{TEST_DATASTORE} --bridge-network=bridge --container-network "vm-network":vmnet ${vicmachinetls}
+    Log  ${output}
+    Should Contain  ${output}  Installer completed successfully
+    Get Docker Params  ${output}  ${true}
+    Log To Console  Installer completed successfully: %{VCH-NAME}
+
+    ${output}=  Run  docker %{VCH-PARAMS} network ls
+    Should Contain  ${output}  vmnet
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} run -d --net=vmnet -p 80 nginx
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+
+    ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} ps
+    Should Not Contain  ${output}  ->80/tcp
+
+    ${rc}  ${output}=  Run And Return Rc And Output  govc vm.power -reset=true %{VCH-NAME}
+    Log To Console  ${output}
+    Should Be Equal As Integers  0  ${rc}
+
+    Reboot VM  %{VCH-NAME}
+
+    Log To Console  Getting VCH IP ...
+    ${new-vch-ip}=  Get VM IP  %{VCH-NAME}
+    Log To Console  New VCH IP is ${new-vch-ip}
+    Replace String  %{VCH-PARAMS}  %{VCH-IP}  ${new-vch-ip}
+
+    # wait for docker info to succeed
+    Wait Until Keyword Succeeds  20x  5 seconds  Run Docker Info  %{VCH-PARAMS}
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} ps
+    Should Not Contain  ${output}  ->80/tcp
+
+    # Delete the portgroup added by env vars keyword
+    Cleanup VCH Bridge Network  %{VCH-NAME}
```

Comment:
Is this going to run if the test fails before this?

Reply:
TBH I'm not 100% sure it's necessary at all?

Reply:
Ok then the question is does the vic-machine delete run on test failure?

Reply:
yes